Make Jackdaw compatible with recent changes to Lara-Framework

lm-sousa opened the pull request:
This PR updates Jackdaw to keep it compatible with specs-feup/lara-framework after the merge of:
- specs-feup/lara-framework#41
- specs-feup/lara-framework#73

It is therefore dependant on its prior merge to build successfully and pass the test suite.

lm-sousa committed:
Update APIs to reflect upstream lara-framework API changes

lm-sousa committed:
Update Language Specification to reflect new conventions

lm-sousa committed:
Delete obsolete API

lm-sousa committed:
Reverts recent LaguageSpecification changes

joaobispo committed:
Adds support for Gradle build system

joaobispo committed:
Fixes build.gradle, moves resource files to their own folder

lm-sousa commented on jsweaver/src/pt/up/fe/specs/jackdaw/abstracts/weaver/AJackdawWeaver.java:
```diff
@@ -8,8 +8,8 @@
  * Abstract Weaver Implementation for JackdawWeaver<br>
- * Since the generated abstract classes are always overwritten, their implementation should be done by extending those abstract classes with user-defined classes.<br>
- * The abstract class {@link pt.up.fe.specs.jackdaw.abstracts.AJackdawWeaverJoinPoint} can be used to add user-defined methods and fields which the user intends to add for all join points and are not intended to be used in LARA aspects.
+Since the generated abstract classes are always overwritten, their implementation should be done by extending those abstract classes with user-defined classes.<br>
+The abstract class {@link pt.up.fe.specs.jackdaw.abstracts.AJackdawWeaverJoinPoint} can be used to add user-defined methods and fields which the user intends to add for all join points and are not intended to be used in LARA aspects.
```
What happened here?

joaobispo replied:
If the question is about the change, it appears to be related to auto-formatting.

About the contents, apparently this message was written before the same class was used to host attributes and actions that are common to all join points, and that can and should be used in LARA aspects. This is outdated, the weaver generator in lara-framework will be updated to change the message so that it better reflects how it is used nowadays.

joaobispo committed:
Just to trigger GitHub action

joaobispo committed:
Merge branch 'staging' of github.com:specs-feup/jsweaver into ts-conversion

joaobispo committed:
Changes lara-js workflow to Gradle instead of Eclipse build

joaobispo committed:
Removes files committed to bin folder

joaobispo committed:
Adds bin folder to .gitignore

joaobispo committed:
Merge branch 'master' of github.com:specs-feup/jsweaver into ts-conversion

joaobispo committed:
Merge branch 'staging' of github.com:specs-feup/jsweaver into ts-conversion

joaobispo committed:
Merge branch 'staging' of github.com:specs-feup/jsweaver into ts-conversion

# Conflicts:
#	.github/workflows/ant-main.yml

joaobispo committed:
Removes workflows not related to this branch

joaobispo committed:
Re-adds workflows, because of PR

- This branch should have only had its own workflow, but now that they have been in the branch, they should stay until the PR;

joaobispo committed:
Trigger workflow run

joaobispo committed:
Merge branch 'master' of github.com:specs-feup/jsweaver into ts-conversion
